akka-stream-tests/src/test/scala/akka/stream/scaladsl/LazyFlowSpec.scala

```diff
@@ -140,44 +135,35 @@ class LazyFlowSpec extends StreamSpec {
       sourceSub.expectCancellation()
     }
 
-    "continue if supervision is resume" in assertAllStagesStopped {
-      val sourceProbe = TestPublisher.manualProbe[Int]()
-      def flowBuilder(a: Int) = if (a == 0) throw ex else Future.successful(Flow.fromFunction[Int, Int](id ⇒ id))
-      val probe = Source.fromPublisher(sourceProbe)
-        .via(Flow.lazyInit[Int, Int, NotUsed](flowBuilder, fallback))
-        .withAttributes(supervisionStrategy(resumingDecider))
-        .runWith(TestSink.probe[Int])
-
-      val sourceSub = sourceProbe.expectSubscription()
-      probe.request(1)
-      sourceSub.expectRequest(1)
-      sourceSub.sendNext(0)
-      sourceSub.expectRequest(1)
-      sourceSub.sendNext(1)
-      probe.expectNext(1)
-      probe.cancel()
-    }
-
-    "fail correctly when materialization of inner sink fails" in assertAllStagesStopped {
-      val matFail = TE("fail!")
-      object FailingInnerMat extends GraphStageWithMaterializedValue[FlowShape[String, String], Option[String]] {
-        val in = Inlet[String]("in")
-        val out = Outlet[String]("out")
-        val shape = FlowShape(in, out)
-        override def createLogicAndMaterializedValue(inheritedAttributes: Attributes) =
-          (new GraphStageLogic(shape) {
-            throw matFail
-          }, Some("fine"))
-      }
-
+    //    "continue if supervision is resume" in assertAllStagesStopped {
+    //      val sourceProbe = TestPublisher.manualProbe[Int]()
+    //      def flowBuilder(a: Int) = if (a == 0) throw ex else Future.successful(Flow.fromFunction[Int, Int](id ⇒ id))
+    //      val probe = Source.fromPublisher(sourceProbe)
+    //        .via(Flow.lazyInit[Int, Int, NotUsed](flowBuilder))
+    //        .withAttributes(supervisionStrategy(resumingDecider))
+    //        .runWith(TestSink.probe[Int])
+    //
+    //      val sourceSub = sourceProbe.expectSubscription()
+    //      probe.request(1)
+    //      sourceSub.expectRequest(1)
+    //      sourceSub.sendNext(0)
+    //      sourceSub.expectRequest(1)
+    //      sourceSub.sendNext(1)
+    //      probe.expectNext(1)
+    //      probe.cancel()
+    //    }
```

Comment:
why? please remove if not valid anymore

Reply:
I did not remove these lines yet because I first wanted to discuss the supervision issue (cf. may initial comment - "There may be one issue left: ...").

@gosubpl 
I would like to remove supervision support / logic from LazyFlow:

First: The new implementation `Flow.lazyInitAsync` no more provides the first element to the `flowFactory` function. If the construction of the inner flow fails then this is caused by "external" factors only. If a retry mechanism is needed then it can be implemented on the Future level by using `Future.recoverWith` or `Future.recover`.

Second: If the materialization of the inner flow fails then there is most likely a logic error in the returned flow blueprint. I do not think that supervision strategies are meant to handle these kinds of failures.